engine/src/execution_graph/logic_controllers/BatchProcessing.h

```diff
@@ -150,6 +150,10 @@ class ExternalBatchColumnDataSequence {
 		t.detach();
 	} 
 
+	bool wait_for_next() {
```

Comment:
does wait_for_next not exist for all cache types?

Reply:
Apparently not... Here we need to do some refactoring. There should be an interface that all BatchSequencers use. I did not realize there was no interface until i found out that wait_for_next did not exist for ExternalBatchColumnDataSequence.
I think there is a couple other classes that should have an interface that dont.
I could implement the interface as part of this PR, but I think this PR is too big as it is. I think we should do it as part of another PR where we locate and implement any missing interfaces

Reply:
Ok make an issue and we can resolve it there. Yes it should have an interface and ideally we should be using the more abstract interface because they all implement the same methods

Reply:
Here is the issue: https://github.com/BlazingDB/blazingsql/issues/573